realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1415,8 +1421,9 @@ private void checkMaxDepth(int maxDepth) {
      * version, nothing will happen.
      *
      * @param configuration {@link RealmConfiguration}
+     * @throws FileNotFoundException if the Realm file doesn't exist.
      */
-    public static void migrateRealm(RealmConfiguration configuration) {
+    public static void migrateRealm(RealmConfiguration configuration) throws FileNotFoundException {
```

Comment:
Do we really want to make this a checked exception? I would say it isn't an expected error, but a logical error in the code path?